src/components/ui/sidebar.tsx

```diff
@@ -112,6 +112,17 @@ const SidebarProvider = React.forwardRef<
       toggleSidebar,
     );
 
+    // load the sidebar state from the cookie
+    React.useEffect(() => {
+      const cookie = document.cookie
+        .split("; ")
+        .find((row) => row.startsWith(SIDEBAR_COOKIE_NAME));
+      if (cookie) {
+        const value = cookie.split("=")[1];
+        setOpen(value === "true");
+      }
+    }, []);
+
```

Comment:
Why don't we follow the docs? 
 https://ui.shadcn.com/docs/components/sidebar#:~:text=Copy-,Persisted%20State,-The%20SidebarProvider%20supports

Reply:
@bodhish using `defaultOpen` adds to code complexity. Also we do not have server side support, so the cookies are loaded client side, with the default open state being closed, resulting in the opening animation on reload. If that is okay, i'll implement that.

Reply:
@bodhish

Reply:
I think that should be fine, we don't have to maintain the logic right!